update_compilers/install_nonfree_compilers.sh

```diff
@@ -36,6 +36,20 @@ for version in 2016.3.210 2018.0.033; do
     fi
 done
 
+install_oneapi() {
+    local HPCKIT_URL=$1
+    local TMP_DIR=${OPT}/tmp/oneapi
+    local HPCKIT_PATH=${TMP_DIR}/hpckit.sh
+    local INSTALL_DIR=$(pwd)/$2
+    mkdir -p ${TMP_DIR}
+    fetch ${HPCKIT_URL} > $HPCKIT_PATH
+    bash $HPCKIT_PATH -s -a -s --action install --eula accept --components intel.oneapi.lin.cpp-compiler:intel.oneapi.lin.ifort-compiler --install-dir ${DIR}
+    # to use the compiler, you must 'source ${DIR}/setvars.sh'
+    rm -rf ${TMP_DIR}
+}
+
+install_oneapi https://registrationcenter-download.intel.com/akdlm/irc_nas/16856/l_HPCKit_b_2021.1.8.1883_offline.sh 2021-beta8
```

Comment:
This is great: how likely is the download to stick around? My experience is that "normally" versioned things like "gcc-9.2.0" the tarballs tend to stay around forever, but relatively unusual seemingly random things like this tend to vanish from download servers randomly.

If the latter is true we may wish to do what we "normally" do; which is to install, then tarball up and save the resulting installation image. This is how the existing intel compilers are installed.

Basically this comes down to a kind of "backup", can we recover if our NFS drive gets zapped my a lightning bolt. Or idiocy on my part :)  Perhaps I'm just paranoid!

Reply:
The URLs do not change. The 5 digit number is the unique identifier for the release in the content management system.

It looks like there is some manual effort for you to post a new compiler. I was hoping we could get to a point where we could just submit PR's when new compilers are available.

Reply:
I'd say let's use this new way (fitted into the installer system and not this script) and blame Intel when things go missing

Reply:
Thanks both: agreed! I think this is a big improvement!

I'm in the process of changing how we install things: I can ensure it works for this too.

Reply:
Re: `# to use the compiler, you must 'source ${DIR}/setvars.sh'` -- we'll have to find a way around this as we don't run a regular shell to do compilations.

Reply:
This works for me:

LD_LIBRARY_PATH=/opt/intel/oneapi/compiler/latest/linux/compiler/lib/intel64_lin/:$LD_LIBRARY_PATH /opt/intel/oneapi/compiler/latest/linux/bin/intel64/icpc  ex.cpp